hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3331,6 +3341,16 @@ func (r *HostedClusterReconciler) reconcileNetworkPolicies(ctx context.Context,
 		}); err != nil {
 			return fmt.Errorf("failed to reconcile virt launcher policy: %w", err)
 		}
+		kvInfraCluster, err := r.discoverKubevirtClusterClient(ctx, hcluster)
+		if err != nil {
+			return err
+		}
+		egressFirewall := egressfirewall.VirtLauncherEgressFirewall(kvInfraCluster.namespace)
```

Comment:
in case of local infra, doesn't it mean the `EgressFirewall` will be created at the hostedcluster namespace (i.e. `clusters` namespace)?
the assignment in that case is being done here:
https://github.com/openshift/hypershift/pull/2399/files#diff-b5cde720842b19b53da3f454d827e27327c4559b2e4849906dac728586f51eb3R4605
i believe we want to create the EgressFirewall where the VMs are, i.e. in the HCP namespace (`clusters-<guest_cluster_name>`)

Reply:
Right

```
$ oc get egressfirewall -A
NAMESPACE   NAME      EGRESSFIREWALL STATUS
clusters    default   EgressFirewall Rules applied
```

I don't now why the others lie Rules can be created at hc namespace.

Reply:
for `kubevirtInfraTempRoute`, it doesn't matter on which namespace it is being created, it is just to extract the base domain, and the the route is deleted.
for the egress firewall, it is important to create it in the same namespace as the VirtualMachines.

Reply:
```
oc get egressfirewall -A
NAMESPACE   NAME      EGRESSFIREWALL STATUS
clusters    default   EgressFirewall Rules applied
```

Is this right?! doesn't the egressfirewall need to be created within the namespace that the VMs run in?

> for the egress firewall, it is important to create it in the same namespace as the VirtualMachines.

right, we shouldn't be putting the egress rule inside the HostedCluster's namespace, but instead the HostedControlPlane's namespace or the external infra namespace

That said, maybe you already fixed this? because i see the hcp.Namespace being used now in the `discoverKubevirtClusterClient` function

Reply:
Is already fixed.